src/components/MAPI/apps/ClusterDetailIngress.tsx

```diff
@@ -45,41 +51,109 @@ interface IClusterDetailIngressProps
 
 const ClusterDetailIngress: React.FC<
   React.PropsWithChildren<IClusterDetailIngressProps>
+  // eslint-disable-next-line complexity
 > = ({
-  provider,
   k8sEndpoint,
   kvmTCPHTTPPort,
   kvmTCPHTTPSPort,
   mutateCluster,
   ...rest
 }) => {
+  const provider = rest.provider ?? window.config.info.general.provider;
   const { pathname } = useLocation();
-  const { clusterId } = useParams<{ clusterId: string }>();
+  const { clusterId, orgId } = useParams<{
+    clusterId: string;
+    orgId: string;
+  }>();
 
+  const clientFactory = useHttpClientFactory();
   const auth = useAuthProvider();
 
-  const appListClient = useHttpClient();
-  const appsPermissions = usePermissionsForApps(
-    provider ?? window.config.info.general.provider,
-    clusterId
+  const orgClient = useHttpClient();
+  const orgKey = securityv1alpha1.getOrganizationKey(orgId);
+  const { data: org, error: orgError } = useSWR<
+    securityv1alpha1.IOrganization,
+    GenericResponseError
+  >(orgKey, () => securityv1alpha1.getOrganization(orgClient, auth, orgId));
+
+  useEffect(() => {
+    if (orgError) {
+      ErrorReporter.getInstance().notify(orgError);
+
+      new FlashMessage(
+        'There was a problem fetching an organization resource.',
+        messageType.ERROR,
+        messageTTL.LONG,
+        extractErrorMessage(orgError)
+      );
+    }
+  }, [orgError]);
+
+  const namespace = org?.status?.namespace;
+
+  const { canGet: canGetClusters } = usePermissionsForClusters(
+    provider,
+    namespace ?? ''
   );
-  const appListGetOptions = { namespace: clusterId };
+
+  const clusterKey =
+    canGetClusters && namespace
+      ? fetchClusterKey(provider, namespace, clusterId)
+      : null;
+
+  const { data: cluster, error: clusterError } = useSWR<
+    Cluster,
+    GenericResponseError
+  >(clusterKey, () =>
+    fetchCluster(clientFactory, auth, provider, namespace!, clusterId)
```

Comment:
In `<ClusterDetailApps />` we get the `isClusterApp` prop from the parent component and get the organization's namespace from the current selected organization. Is it also possible to do the same here, or is there another reason for the different implementation?

Reply:
Good idea, should work